app/util/networks/handleNetworkSwitch.ts

```diff
@@ -1,33 +1,35 @@
-import type { NetworkController } from '@metamask/network-controller';
+import { CurrencyRateController } from '@metamask/assets-controllers';
+import { NetworkType } from '@metamask/controller-utils';
+import { NetworkController } from '@metamask/network-controller';
 import { getNetworkTypeById } from './index';
-import type { CurrencyRateController } from '@metamask/assets-controllers';
-
-const handleNetworkSwitch = (
-  switchToChainId: string,
-  {
-    networkController,
-    currencyRateController,
-  }: {
-    networkController: NetworkController;
-    currencyRateController: CurrencyRateController;
-  },
-): string | undefined => {
+import Engine from '../../core/Engine';
+
+/**
+ * Switch to the given chain ID.
+ *
+ * @returns The network type of the build-in network switched to, or the
+ * network ID of the custom network switched to, or undefined if no switch
+ * occurred.
+ */
+const handleNetworkSwitch = (switchToChainId: string): string | undefined => {
   // If not specified, use the current network
   if (!switchToChainId) {
     return;
   }
 
+  const currencyRateController = Engine.context
+    .CurrencyRateController as CurrencyRateController;
+  const networkController = Engine.context
+    .NetworkController as NetworkController;
+
   // If current network is the same as the one we want to switch to, do nothing
-  if (
-    networkController?.state?.providerConfig?.chainId ===
-    String(switchToChainId)
-  ) {
+  if (networkController.state.providerConfig.chainId === switchToChainId) {
```

Comment:
Should we keep the `String` conversion in here in case `switchToChainId` is still a number for some reason?

Reply:
Great catch, I had meant to review this to see if this was safe. I don't think it is, I found a couple of weird cases at least. I'll restore this.

```suggestion
  if (networkController.state.providerConfig.chainId === String(switchToChainId)) {
```